### PR TITLE
[WIP] Rewrite WebGLRenderingContext to use typed array APIs 

### LINKED ISSUE #14676: Rewrite WebGLRenderingContext to use typed array APIs
We need to replace uses of `array_buffer_view_data_checked`, `array_buffer_view_data`, `array_buffer_view_to_vec_checked`, `array_buffer_view_to_vec`, and `array_buffer_to_vec`. This may mean adding an additional `ArrayBufferViewU16` type to the list in rust-mozjs, but the rest should be straightforward replacements with things like `ArrayBuffer::from` and `ArrayBufferView::from` (after https://github.com/servo/rust-mozjs/pull/323 merges).

Code: `components/script/dom/webglrenderingcontext.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/webgl/`

### PR DESCRIPTION
**Pull request by yysung1123**
---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #14676 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15143)

### CODE REVIEW

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -436,7 +436,8 @@ impl WebGLRenderingContext {
                                          height: u32,
                                          format: TexFormat,
                                          data_type: TexDataType,
-                                         data: *mut JSObject)
+                                         data: *mut JSObject
```
Comment (stdio2016) [inline components/script/dom/webglrenderingcontext.rs]: 是不是少了逗號在最後面？

Comment (stdio2016) [inline components/script/dom/webglrenderingcontext.rs]: missing comma at the end of line?

**Comment by jdm** [comment]
@deror1869107 What was the purpose of opening this PR right now?

File: components/script/dom/webglrenderingcontext.rs (hunk 1 above)
Comment (deror1869107) [inline components/script/dom/webglrenderingcontext.rs]: Yes. I forgot it.

**Comment by deror1869107** [comment]
@jdm 
#14676

**Comment by jdm** [comment]
@deror1869107 I know it's associated with #14676, but currently it adds a single argument to a function that does not get used :)

**Comment by deror1869107** [comment]
So I should only PR after finishing the issue.
Sorry for that. I'll PR again after finishing #14676 .

**Comment by jdm** [comment]
Yeah, we usually open PRs when the changes are ready for review, or when the changes are incomplete but we want to ask a specific question about some changes.

**Pull request closed without merging**